pom.xml

```diff
@@ -21,6 +21,61 @@
                     <target>1.7</target>
                 </configuration>
             </plugin>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-jar-plugin</artifactId>
+				<configuration>
+					<archive>
+						<addMavenDescriptor>false</addMavenDescriptor>
+						<manifest>
+							<addDefaultSpecificationEntries>true</addDefaultSpecificationEntries>
+							<addDefaultImplementationEntries>true</addDefaultImplementationEntries>
+						</manifest>
+					</archive>
+				</configuration>
+				<version>2.4</version>
+			</plugin>
+			<plugin>
+				<groupId>org.bsc.maven</groupId>
+				<artifactId>maven-processor-plugin</artifactId>
+				<version>2.2.4</version>
+				<executions>
+					<execution>
+						<id>process</id>
+						<phase>process-classes</phase>
+						<goals>
+							<goal>process</goal>
+						</goals>
+					</execution>
+				</executions>
+				<configuration>
+					<outputDirectory>src/main/generated</outputDirectory>
+					<processors>
+						<processor>com.laytonsmith.core.extensions.ExtensionAnnotationProcessor</processor>
+					</processors>
+				</configuration>
+			</plugin>
+			<plugin>
+				<groupId>org.codehaus.mojo</groupId>
+				<artifactId>exec-maven-plugin</artifactId>
+				<version>1.2.1</version>
+				<executions>
+					<execution>
+						<id>cache-annotations</id>
+						<phase>process-classes</phase>
+						<goals>
+							<goal>java</goal>
+						</goals>
+					</execution>
+				</executions>
+				<configuration>
+					<mainClass>com.laytonsmith.PureUtilities.ClassLoading.Annotations.CacheAnnotations</mainClass>
+					<arguments>
+						<argument>${basedir}/target/classes</argument>
+						<argument>${basedir}/target/classes</argument>
+					</arguments>
+				</configuration>
+			</plugin>
```

Comment:
It's in the extension skeleton. I'm not sure what the first plugin is doing, so you could look into that, but the second "verifies some things so that your extension works" and the third "speeds up extension loading". I added these when I was trying to get it to build in my IDE; turns out it was an unrelated problem, but I thought the extension would benefit from the last two at least, so I left it in.